pkg/sqlite/load_test.go

```diff
@@ -825,3 +820,211 @@ func TestAddBundlePropertiesFromAnnotations(t *testing.T) {
+					"pkg-1/b/csv-b": {},
+				},
+			},
+		},
+		{
+			description: "ChannelHead/WithReplacement",
+			fields: fields{
+				bundles: []*registry.Bundle{
+					newBundle(t, "csv-a", "pkg-0", []string{"a", "b"}, newUnstructuredCSV(t, "csv-a", "")),
+					newBundle(t, "csv-aa", "pkg-0", []string{"b"}, newUnstructuredCSV(t, "csv-aa", "csv-a")),
+				},
+				pkgs: []registry.PackageManifest{
+					{
+						PackageName: "pkg-0",
+						Channels: []registry.PackageChannel{
+							{
+								Name:           "a",
+								CurrentCSVName: "csv-a",
+							},
+							{
+								Name:           "b",
+								CurrentCSVName: "csv-aa",
+							},
+						},
+						DefaultChannelName: "b",
+					},
+				},
+			},
+			args: args{
+				bundle: "csv-a",
+				pkg:    "pkg-0",
+			},
+			expected: expected{
+				err: fmt.Errorf("cannot overwrite bundle csv-a from package pkg-0: replaced by csv-aa on channel b"),
+				bundles: map[string]struct{}{
+					"pkg-0/a/csv-a":  {},
+					"pkg-0/b/csv-a":  {},
+					"pkg-0/b/csv-aa": {},
+				},
+			},
+		},
+		{
+			description: "ChannelHead",
+			fields: fields{
+				bundles: []*registry.Bundle{
+					newBundle(t, "csv-a", "pkg-0", []string{"a", "b"}, newUnstructuredCSVWithSkips(t, "csv-a", "csv-b", "csv-c")),
+					newBundle(t, "csv-b", "pkg-0", []string{"b", "d"}, newUnstructuredCSV(t, "csv-b", "")),
+					newBundle(t, "csv-d", "pkg-0", []string{"d"}, newUnstructuredCSV(t, "csv-d", "csv-b")),
+					newBundle(t, "csv-c", "pkg-0", []string{"c"}, newUnstructuredCSV(t, "csv-c", "")),
+				},
+				pkgs: []registry.PackageManifest{
+					{
+						PackageName: "pkg-0",
+						Channels: []registry.PackageChannel{
+							{
+								Name:           "a",
+								CurrentCSVName: "csv-a",
+							},
+							{
+								Name:           "b",
+								CurrentCSVName: "csv-a",
+							},
+							{
+								Name:           "c",
+								CurrentCSVName: "csv-c",
+							},
+							{
+								Name:           "d",
+								CurrentCSVName: "csv-d",
+							},
+						},
+						DefaultChannelName: "a",
+					},
+				},
+			},
+			args: args{
+				bundle: "csv-a",
+				pkg:    "pkg-0",
+			},
+			expected: expected{
+				err: nil,
+				bundles: map[string]struct{}{
+					"pkg-0/a/csv-b": {},
+					"pkg-0/b/csv-b": {},
+					"pkg-0/a/csv-c": {},
+					"pkg-0/b/csv-c": {},
+					"pkg-0/c/csv-c": {},
+					"pkg-0/d/csv-d": {},
+					"pkg-0/d/csv-b": {},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			db, cleanup := CreateTestDb(t)
+			defer cleanup()
+			store, err := NewSQLLiteLoader(db)
+			require.NoError(t, err)
+			err = store.Migrate(context.TODO())
+			require.NoError(t, err)
+
+			for _, bundle := range tt.fields.bundles {
+				// Throw away any errors loading bundles (not testing this)
+				store.AddOperatorBundle(bundle)
+			}
+
+			for _, pkg := range tt.fields.pkgs {
+				// Throw away any errors loading packages (not testing this)
+				store.AddPackageChannels(pkg)
```

Comment:
Do we expect errors to occur? If not, can we `require.NoError` each of them. If we do expect errors, do we know which ones?

It worries me a little that errors happening here may cause false positives or false negatives if we later make changes in the code under test.

Reply:
switching it to `require.NoError` seems trivial and the right thing to do